client/src/main/java/org/eclipse/hono/client/Command.java

```diff
@@ -381,10 +418,12 @@ public String toString() {
      * @param deviceId The device id.
      * @return The reply-to-id, starting with the device id.
      */
-    static String getDeviceFacingReplyToId(final String replyToId, final String deviceId) {
-        if (replyToId.startsWith(deviceId + "/")) {
-            return replyToId.replaceFirst(deviceId + "/", deviceId + "/0");
+    public static String getDeviceFacingReplyToId(final String replyToId, final String deviceId) {
+        final boolean replyToContainedDeviceId = replyToId.startsWith(deviceId + "/");
+        final String bitFlagString = encodeReplyToOptions(replyToContainedDeviceId);
+        if (replyToContainedDeviceId) {
+            return replyToId.replaceFirst(deviceId + "/", deviceId + "/" + bitFlagString);
```

Comment:
if I understand the code correctly, then you are now adding a `1` in case the reply-to-id contains the device-id instead of adding a `0` as it was before. Is this so that you are able to use the same *encode*/*decode* methods?

Reply:
Yes, that's the intention (see also PR description above).